pkg/utils/elasticsearch7.go

```diff
@@ -360,3 +374,128 @@ func (es *Elasticsearch7) DeleteTemplate(ctx context.Context, templateName strin
 	es.log.Info("template was deleted successfully", "templateName", templateName)
 	return nil
 }
+
+func (es *Elasticsearch7) DeletePipeline(ctx context.Context, pipelineName string) error {
+	ctx, cancel := context.WithTimeout(ctx, ElasticMainFnTimeout)
+	defer cancel()
+
+	exists, err := es.existsPipeline(ctx, pipelineName)
+	if err != nil {
+		es.log.Error(err, "pipeline cannot be deleted", "pipelineName", pipelineName)
+		return err
+	}
+	if !exists {
+		es.log.Info("pipeline cannot be deleted because it does not exists", "pipelineName", pipelineName)
+		return nil
+	}
+
+	pipelineStatus, err := es.isPipelineUsed(ctx, pipelineName)
```

Comment:
Keep it simple: try delete, if error => return the error to the caller method...
No need to go deeper and check if Pipeline is used by an Index, because Elasticsearch will check this and report the error later when you execute `esapi.IngestDeletePipelineRequest`.
So I think you can delete the `isPipelineUsed` method.

Reply:
done